operators/rx_test.go

```diff
@@ -38,6 +38,12 @@ func TestRx(t *testing.T) {
 			input:   "グッバイワールド",
 			want:    false,
 		},
+		// Requires dotall
```

Comment:
I couldn't come up with a simplified unit test for requiring multiline. The failing one is

https://github.com/coreruleset/coreruleset/blob/v4.0/dev/rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf#LL1507C37-L1507C37

I tried using `^` and `\b` together but it didn't seem to fail without the `m` option. Any other regex gurus able to suggest a simple unit test that fails without `m`?

Reply:
This test seems not working without `m`:
```
// Requires multiline
{
	pattern: `^hello.*world`,
	input:   "test\nhello\nworld",
	want:    true,
},
```
```
--- FAIL: TestRx (0.00s)
    --- FAIL: TestRx/^hello.*world/test_hello_world (0.00s)
        /Users/matteopace/Repo/coraza/operators/rx_test.go:71: want true, got false
```

Reply:
Good one.